backend/test_pattern_debug.py

```
@@ -0,0 +1,105 @@
+    merchant_info = extractor.extract_merchant_from_description(test_description)
+    if merchant_info:
+        print(f"Merchant detected: {merchant_info.name} with patterns: {merchant_info.common_patterns}")
+    else:
+        print("No merchant detected")
+    
+    print()
+    print("=== Testing Rule Matching Logic ===")
+    
+    # Test how the rule engine would match these patterns
+    from services.category_rule_engine import CategoryRuleEngine
+    from models.category import CategoryRule, MatchCondition
```

Ruff (0.11.9)

`models.category.MatchCondition` imported but unused

Remove unused import: `models.category.MatchCondition`

(F401)